### DISCUSSION TITLE
Hello

### DISCUSSION THREAD

**Discussion by hugheb** [GMOD/jbrowse-components#4069] [General]
I tried to JBrowse2 to align two plant genomes (1 Gb). I got the following error
Error: Cannot create a string longer than 0x1fffffe8 characters

**Comment by cmdcolin** [comment]
Hi there,
How did you create the alignment? And how large was the resulting output file? Is it PAF format?

JBrowse 2 does have to load the entire file into memory, and large alignments can exceed browser limits (1fffffe8->~532 megabytes)

Even if they can be loaded, it may be slow. I would recommend possibly filtering out small alignments. 

You can also consider using a MCScan type alignment (just the protein sequences) instead of a whole genome alignment, as the datasets for MCScan are much smaller and much less tripped up by repetitive sequences (https://github.com/tanghaibao/jcvi/wiki/MCscan-(Python-version))


Down the line, we are working on some things to make loading large whole genome alignments better by using indexing. If you are interested in trying that out as a beta branch I can try to help describe more. That work is here https://github.com/GMOD/jbrowse-components/pull/3859

**Comment by hugheb** [comment]
Dear Colin

I installed JBrowse2 on my computer (MacBook pro, 2.3 GHz 18-Core Intel
Xeon W; 256 GB 2666 MHz DDR4)
I uploaded the reference genome as a fasta file (964.2 Mb) and a draft
genome of another accession of the same species (fasta file size is 1Gb). I
used the Type as IndexedFastaAdapter which is default. I want to find out
what scaffolds from the draft genome match chromosome 7 of my reference
genome.
would it make a difference if I use BgzipFastaAdapter as you recommended in
your presentation at PAG2022.
This is my first attempt at JBrowse.

kind regards

Hossein

**Reply by cmdcolin** [reply]
are you using JBrowse desktop? can you provide a screenshot of the error?

**Comment by hugheb** [comment]
Hello Colin

From your response, it seems JBrowse is for displaying alignment. If this
is the case then I have got it all wrong. I was trying to do the alignment
using an assembled reference and a draft genome.
attached is the screenshot of the error message.

thank you

**Reply by cmdcolin** [reply]
I don't think the screenshot properly attached to your post

**Comment by hugheb** [comment]
attached is a PDF. hopefully it works this time

thanks

**Reply by cmdcolin** [reply]
sorry I think it doesn't work if you attach the file in an email reply. You can reply in the github page or send me a attachment via email at colin.diesh@gmail.com
